server/src/main/java/org/opensearch/wlm/QueryGroupTask.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.tasks.TaskId;
+import org.opensearch.tasks.CancellableTask;
+
+import java.util.Map;
+
+import static org.opensearch.search.SearchService.NO_TIMEOUT;
+
+/**
+ * Base class to define QueryGroup tasks
+ */
+public class QueryGroupTask extends CancellableTask {
```

Comment:
@jainankitk @kaushalmahi12 - CancellableTask is extended by all these classes (image attached)
since we want to focus only on search requests should we scope this to SearchTasks and SearchShardTasks ?

![SCR-20240725-lksu](https://github.com/user-attachments/assets/7cd2fa89-dedd-476f-8ea6-c0f5bb9fc47a)

Reply:
@kiranprakash154 - Why do you think it is not scoped to SearchTasks and SearchShardTasks? Even though we are extending `CancellableTask`, none of the other classes are extending from it. So, how does it impact other implementations of `CancellableTask`?

Reply:
i got confused, i see the changes in SearchShardTask and SearchTask and it makes sense